modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -23,6 +37,292 @@
+            num_row_groups,
+            NPartitions.get(),
+            min_block_size=1,
+        )
+        current_partition_size = 0
+        file_index = 0
+        partition_files = []  # 2D array - each element contains list of chunks to read
+        row_groups_used_in_current_file = 0
+        total_row_groups_added = 0
+        # On each iteration, we add a chunk of one file. That will
+        # take us either to the end of a partition, or to the end
+        # of a file.
+        while total_row_groups_added < num_row_groups:
+            if current_partition_size == 0:
+                partition_files.append([])
+            partition_file = partition_files[-1]
+            file_path = parquet_files[file_index]
+            row_group_start = row_groups_used_in_current_file
+            row_groups_left_in_file = (
+                row_groups_per_file[file_index] - row_groups_used_in_current_file
+            )
+            row_groups_left_for_this_partition = step - current_partition_size
+            if row_groups_left_for_this_partition <= row_groups_left_in_file:
+                # File has at least what we need to finish partition
+                # So finish this partition and start a new one.
+                num_row_groups_to_add = row_groups_left_for_this_partition
+                current_partition_size = 0
+            else:
+                # File doesn't have enough to complete this partition. Add
+                # it into current partition and go to next file.
+                num_row_groups_to_add = row_groups_left_in_file
+                current_partition_size += num_row_groups_to_add
+            if num_row_groups_to_add == row_groups_left_in_file:
+                file_index += 1
+                row_groups_used_in_current_file = 0
+            else:
+                row_groups_used_in_current_file += num_row_groups_to_add
+            partition_file.append(
+                ParquetFileToRead(
+                    file_path, row_group_start, row_group_start + num_row_groups_to_add
+                )
+            )
+            total_row_groups_added += num_row_groups_to_add
+
+        assert (
+            total_row_groups_added == num_row_groups
+        ), "row groups added does not match total num of row groups across parquet files"
+
+        all_partitions = []
+        for files_to_read in partition_files:
+            all_partitions.append(
+                [
+                    cls.deploy(
+                        cls.parse_fsspec_files,
+                        files_for_parser=files_to_read,
+                        columns=cols,
+                        num_returns=3,
+                        storage_options=storage_options,
+                        **kwargs,
+                    )
+                    for cols in col_partitions
+                ]
+            )
+        return all_partitions
+
+    @classmethod
+    def build_partition(cls, partition_ids, column_widths):
```

Comment:
Can we move this method to the parent class (ColumnStoreDispatcher)? Please look at other newly added methods here, can they be moved to the parent?

Reply:
I don't think this specific method can be moved back to the parent because of the way the list of partitions is handled here.